webrecorder/browsertrix
# Initial list of crawls

(WIP https://github.com/webrecorder/browsertrix-cloud/issues/106)

Initial read-only crawl list.

TODO in follow up PRs:
- Show crawl template and user name instead of IDs (needs API update)
- Filtering, sorting
- Poll against running crawls, and some other way of syncing view ("refresh" button for now?)
- Additional stats & file details

### Manual testing
1. Run app and go to "Archives" > archive > "Crawl Templates"
2. Click "Run now" to generate a crawl
3. Click "Crawls" tab. Verify that running crawl is visible at the top (note: will not update without view refresh)
4. Wait for crawl to finish, or verify data with existing finished crawls: crawl ID, crawl template ID, crawl state, completion relative time, start method and start date
5. Run another crawl if there are not multiple crawls and wait for it to finish. Verify that crawls are ordered by start date by default (newest at top)

### Screenshots
**List with running, complete and failed crawls:**
<img width="824" alt="Screen Shot 2022-01-27 at 1 13 58 PM" src="https://user-images.githubusercontent.com/4672952/151444760-1d2e97df-374c-47fe-9d37-b35d7bffc869.png">
